packages/core/src/ceramic.ts

```diff
@@ -179,12 +179,13 @@ class Ceramic implements CeramicApi {
   }
 
   /**
-   * Find handler by genesis record
+   * Find doctype handler
+   * @param doctype - Name of doctype
    * @param genesisRecord - Document genesis record
    */
-  findHandler<T extends DoctypeHandler<Doctype>>(genesisRecord: Record<string, any>): T {
-    if (genesisRecord.doctype in this._doctypeHandlers) {
-      return this._doctypeHandlers[genesisRecord.doctype] as T
+  findHandler<T extends DoctypeHandler<Doctype>>(doctype: string, genesisRecord: Record<string, any>): T {
+    if (doctype in this._doctypeHandlers) {
+      return this._doctypeHandlers[doctype] as T
     } else if (genesisRecord['@context'] === "https://w3id.org/did/v1") {
       return this._doctypeHandlers['3id'] as T
     }
```

Comment:
I think this method can be removed since we would only get the handler from the doctype now. 
`genesisRecord['@context'] === "https://w3id.org/did/v1"` was an old approach to 3ID backwards compatibility which can be removed now.